src/java/src/main/java/com/dnanexus/DXHTTPRequest.java

```diff
@@ -362,48 +361,21 @@ private ParsedResponse requestImpl(String resource, String data, boolean parseRe
                     throw DXAPIException.getInstance(errorType, errorMessage, statusCode);
                 } else {
                     // Propagate 500 error to caller
-                    if (this.disableRetry && statusCode != 503) {
```

Comment:
@geetduggal  or @commandlinegirl can you please review? Not sure why 503 was excluded..

Reply:
The existing 503 logic seems fine if you imagine the typical cmd line dx-toolkit operation. But for TIP, it makes less sense, and it's better to quit after the small # of retries and let spark handle it. Just my opinion.

Reply:
It was added at a client's request  https://jira.internal.dnanexus.com/browse/PTFM-15849. It was actually implemented by adding the ability to disable retries for all 5xx errors (https://github.com/dnanexus/dx-toolkit/commit/968fdd67433dcea7c6b7eb14ce2536991829b9ab) with some special handling of 503.